uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -587,5 +529,80 @@ define([
         }).value;
       }
     });
+
+    describe('roleOf()', function () {
+
+      var promise,
+        defer;
+
+      afterEach(function () {
+        $rootScope.$apply();
+      });
+
+      describe('when contact_id of leave request is same as contact id of parameter', function () {
+
+        beforeEach(function () {
+          spyOn(LeaveRequestAPI, 'isManagedBy').and.callThrough();
+          //dummy contact id
+          LeaveRequestInstance.contact_id = "101";
+          promise = LeaveRequestInstance.roleOf({
+            id: '101'
+          });
+        });
+
+        it('returns owner', function () {
+          promise.then(function (result) {
+            expect(result).toBe('owner');
+          });
+        })
+      });
+
+      describe('when contact_id of leave request is not same as contact id of parameter', function () {
+
+        describe('when isManagedBy return true', function () {
+
+          beforeEach(function () {
+            spyOn(LeaveRequestAPI, 'isManagedBy').and.callFake(function () {
```

Comment:
This is a good example of some setup busywork that you can encapsulate in a function, given that it's not really meaningful as to understand the context of the test

```js
describe('roleOf()', function () {
  describe('when the user is the owner', function () {
    beforeEach(function () {
      // ...
    })

    // ...
  });

  describe('when the user is the manager', function () {
    beforeEach(function () {
      setIsManagedyResponseTo(true);
      // ...
    })

    // ...
  });

  describe('when the user has no relationship', function () {
    beforeEach(function () {
      setIsManagedyResponseTo(false);
      // ...
    })

    // ...
  });

  function setIsManagedyResponseTo(response) {
    spyOn(LeaveRequestAPI, 'isManagedBy').and.callFake(function () {
      return $q.resolve(response);
    });
  }
});
```